src/components/core/geoChart/geoChartDataLayers.ts

```diff
@@ -11,51 +12,66 @@ import {
+import { IGeoData, IGeoPointsConfig, IGeoConfig } from "../../../interfaces/GeoChart";
 import { SEGMENT } from "../../../constants/bucketNames";
 import { getMinMax } from "../../../helpers/utils";
 
 function getExpressionByBucketName(name: string): mapboxgl.Expression {
     return ["get", "value", ["object", ["get", name]]];
 }
 
-function createPushpinSizeOptions(geoData: IGeoData): mapboxgl.Expression | number {
+function createPushpinSizeOptions(
+    geoData: IGeoData,
+    geoPointsConfig: IGeoPointsConfig,
+): mapboxgl.Expression | number {
     const { size } = geoData;
     const hasSize = size !== undefined;
+    const defaultRadius = PUSHPIN_SIZE_OPTIONS_MAP.min.default / 2;
 
     if (!hasSize || size.data.length === 0) {
-        return DEFAULT_PUSHPIN_SIZE_VALUE;
+        return defaultRadius;
     }
 
-    const { min: sizeMin, max: sizeMax } = getMinMax(size.data);
-    if (sizeMax === sizeMin) {
-        return DEFAULT_PUSHPIN_SIZE_VALUE;
+    const { min: minSizeFromData, max: maxSizeFromData } = getMinMax(size.data);
+    if (maxSizeFromData === minSizeFromData) {
+        return defaultRadius;
     }
 
-    const sizesCount = DEFAULT_PUSHPIN_SIZE_SCALE.length;
-    const sizeStep = (sizeMax - sizeMin) / sizesCount;
+    const { minSize: minSizeFromConfig = "default", maxSize: maxSizeFromConfig = "default" } =
+        geoPointsConfig || {};
+    const minSizeInPixel = PUSHPIN_SIZE_OPTIONS_MAP.min[minSizeFromConfig];
+    const maxSizeInPixel = PUSHPIN_SIZE_OPTIONS_MAP.max[maxSizeFromConfig];
+    const base = Math.pow(maxSizeInPixel / minSizeInPixel, 0.25);
+    const getStopPointSize = partial(getPointSize, minSizeInPixel, base);
 
     // The mouseenter event uses queryRenderedFeatures to determine when the mouse is touching a feature
     // And queryRenderedFeatures is not supporting nested object in expression
     // https://github.com/mapbox/mapbox-gl-js/issues/7194
-    const sizeOptions: mapboxgl.Expression = ["step", ["get", "pushpinRadius"], DEFAULT_PUSHPIN_SIZE_VALUE];
-    for (let index = 0; index < sizesCount; index++) {
-        let stepValue = sizeMin + sizeStep * index;
-        stepValue = parseFloat(stepValue.toFixed(2));
-        if (stepValue > sizeMax) {
-            stepValue = sizeMax;
-        }
-        sizeOptions.push(stepValue, DEFAULT_PUSHPIN_SIZE_SCALE[index]);
-    }
-    return ["case", ["==", ["get", "pushpinRadius"], null], DEFAULT_PUSHPIN_SIZE_VALUE, sizeOptions];
+    return [
+        "step",
+        ["get", "pushpinSize"],
+        minSizeInPixel, // a
+        getStopPointSize(1),
+        getStopPointSize(1) / 2, // ar^1
+        getStopPointSize(2),
+        getStopPointSize(2) / 2, // ar^2
+        getStopPointSize(3),
+        getStopPointSize(3) / 2, // ar^3
+        maxSizeInPixel,
+        maxSizeInPixel, // ar^4
+    ];
+}
+
+function getPointSize(minSizeOptionInPixel: number, base: number, exponent: number): number {
```

Comment:
for consistency, I suggest to change `minSizeOptionInPixel`  => `minSizeInPixel `